### ISSUE TITLE
Two consecutive beginTransaction+commitTransaction calls results in an error on Android.

### ISSUE THREAD

**Issue by samwillis** [capacitor-community/sqlite#478]
**Describe the bug**

If you run two transactions using `beginTransaction()` & `commitTransaction()` consecutively, the second `beginTransaction` will throw an error on Android (it works fine on iOS):

> Error: BeginTransaction: Failed in beginTransactionAlready in transaction

**To Reproduce**

I have made a minimal reproduction in this repo: https://github.com/samwillis/capacitor-sqlite-transaction-bug-reproduction

This is a basic Ionic app with a single button to run the test found here: https://github.com/samwillis/capacitor-sqlite-transaction-bug-reproduction/blob/main/src/testSQLite.ts

1. Check it out the repo above
2. Run `ionic capacitor build android`
3. Run the simulator
4. Attach the Chrome dev tools to see the logs
5. Click the test button in the app and observe the console

**Expected behavior**

I would have expected the second `beginTransaction` to succeed as we committed the first one.

**Screenshots**

<img width="1197" alt="image" src="https://github.com/capacitor-community/sqlite/assets/31130/8a159629-12aa-48c4-b201-fe31d0e86cbb">

**Smartphone:**

- Device: Pixel_3a_API_34_extension_level_7_arm64-v8a simulator
- OS: Android 14

**Additional context**

This was found while building a demo combining Capacitor SQLite and [ElectricSQL](http://electric-sql.com) for sync.

A little digging, it seems the error is being thrown here after a check that there isn't a current transaction:

https://github.com/capacitor-community/sqlite/blob/2d7f72bb6a0ebd19042f78e322f258cfb9caa98d/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/Database.java#L153-L158

**Comment by abrunetta** [comment]
Same problem here. Looking at the code it seems that a call to Android's `endTransaction()` is missing after [this line](https://github.com/capacitor-community/sqlite/blob/2d7f72bb6a0ebd19042f78e322f258cfb9caa98d/android/src/main/java/com/getcapacitor/community/database/sqlite/SQLite/Database.java#L182C33-L182C33) in the `Database.commitTransaction()` method in this plugin.

In other words, commits are marked as successful but the transaction is not ended correctly as required by Android.

Please check the [AndroidX documentation](https://developer.android.com/reference/androidx/sqlite/db/SupportSQLiteDatabase#beginTransaction()) on how a `beginTransaction()` should end with calls to both `setTransactionSuccessful()` and `endTransaction()`.

**Comment by jepiqueau** [comment]
@samwillis @abrunetta Thanks for reporting this. Will be fixed in the next release
In waiting you can do this
```
    /**
     * CommitTransaction method
     *
     * @return commit a database transaction
     */
    public Integer commitTransaction() throws Exception {
        if (_db.isOpen()) {
            try {
                if (!isAvailTrans()) {
                    throw new Exception("No transaction active");
                }
                _db.setTransactionSuccessful();
                return 0;
            } catch (Exception e) {
                String msg = "Failed in commitTransaction" + e.getMessage();
                Log.v(TAG, msg);
                throw new Exception(msg);
            } finally {
                _db.endTransaction();
            }

        } else {
            throw new Exception("Database not opened");
        }
    }

    /**
     * Rollback Transaction method
     *
     * @return rollback a database transaction
     */
    public Integer rollbackTransaction() throws Exception {
        if (_db.isOpen()) {
            try {
                if (isAvailTrans()) {
                    _db.endTransaction();
                }
                return 0;
            } catch (Exception e) {
                String msg = "Failed in rollbackTransaction" + e.getMessage();
                Log.v(TAG, msg);
                throw new Exception(msg);
            } finally {
                _db.endTransaction();
            }
        } else {
            throw new Exception("Database not opened");
        }
    }

```

**Comment by jepiqueau** [comment]
@samwillis @abrunetta sorry you do not need to modify the rollbackTransaction method only the commitTransaction method

**Comment by jepiqueau** [comment]
@samwillis @abrunetta fixed in 5.4.1
